cohttp-async/src/server.mli

```diff
@@ -1,3 +1,18 @@
+module Input_channel : sig
+  type t
+
+  val read_line : t -> string option Async_kernel.Deferred.t
+  val read : t -> int -> string Async_kernel.Deferred.t
+  val refill : t -> [> `Eof | `Ok ] Async_kernel.Deferred.t
+
+  val with_input_buffer :
+    t -> f:(string -> pos:int -> len:int -> 'a * int) -> 'a
+
+  val is_closed : t -> bool
+  val close : t -> unit Async_kernel.Deferred.t
+  val close_finished : t -> unit Async_kernel.Deferred.t
+end
```

Comment:
Could you add some docs to this module?

Reply:
Once #842 is merged, this module will be removed as we won't need to access the Input_channel module outside of cohttp-async again.

Reply:
It is now merged.

Reply:
I've synced with the changes from #842 and the diff now only contains the 2 additional functions that will allow using the library with user provided reader/writer